### ISSUE TITLE
Smoothing and threshold methods, current version of the program

### ISSUE THREAD

**Issue by d4fe** [4lex4/scantailor-advanced#110]
1. Where can I read about Savitsky-Golei smoothing and morphological smoothing so that they can be applied consciously?
2. Where can I read about the methods of “Otsu”, “Sauvola”, “Wolf” so that they are consciously applied?
3. Is there a current version of the program with the changes? Or only from August 16, 2019?
(Automatic translation).

**Comment by zvezdochiot** [comment]
@d4fe say:
>"Otsu”, “Sauvola”, “Wolf"

Do you need documentation or clean algorithms?

PS: "Wolf" aka "Chistian".

**Comment by d4fe** [comment]
I need to understand how they process the image in order to apply / not apply when using the program.

**Comment by Piolie** [comment]
A good place to start is Wikipedia: [Savitzky–Golay filter](https://en.wikipedia.org/wiki/Savitzky%E2%80%93Golay_filter), [Otsu's method](https://en.wikipedia.org/wiki/Otsu%27s_method). Also check the references on each article.

Personally, for the binarization step I go with the default (`Otsu`) unless the result seems unsatisfactory. In that case I try changing the settings. Guess the "best" result will always be a subjective matter. I'm glad STA provides different options.

**Comment by d4fe** [comment]
Understood thanks!

**Comment by jeremydmoore** [comment]
I've found Wolf to work quite well for many of the documents I'm processing right now. Even without knowing what each Algorithm does, just pick some sample pages and save the output for each algorithm and pick the one you like the most. Then start doing the same tweaking the settings, just change one setting at a time. We're OCRing our materials and the harder edges of the Wolf algorithm has given me a bit better result.

**Comment by d4fe** [comment]
I agree. This type of anti-aliasing handles fine details better, I work only with it. Other species in these places chop off subtle details.

**Comment by daaaaaaaaaniel** [comment]
Would be very helpful for the README to give basic explanations of how to use the “Otsu”, “Sauvola” and “Wolf” algorithms for non-technical users.

**Comment by cockfighter** [comment]
> Would be very helpful for the README to give basic explanations of how to use the “Otsu”, “Sauvola” and “Wolf” algorithms for non-technical users.

This; what is _Window size_ and how is it applied to output (or interface)?
default value - _window size: 200_

**Comment by l-4-l** [comment]
> > Would be very helpful for the README to give basic explanations of how to use the “Otsu”, “Sauvola” and “Wolf” algorithms for non-technical users.
> 
> This; what is _Window size_ and how is it applied to output (or interface)? default value - _window size: 200_

I think the Window size is one of parameters of the filter (smoothing algorithm is an implementation of a digital image processing filter). Usually it is a size in pixels of a square which is used in process of filtering. With easy words, it's a size of a blur in pixels.

**Comment by jeremydmoore** [comment]
I know this information and have taught it before to non-technical using Python and computer vision. I trained my employees how to use ScanTailor, but I never made any educational resources for it.  I want to help, but I got injured a few years ago that makes everything painful, slow, and frustrating. For example, it has taken me 10 minutes to write this email so far.If you are willing to do the grunt work, I can slowly teach you the algorithmic theories, how to test their use in ScanTailor, and help craft approachable documentation.My best,Jeremy Sent from my iPhoneOn Feb 16, 2024, at 12:34 AM, Andrey B. ***@***.***> wrote:﻿


Would be very helpful for the README to give basic explanations of how to use the “Otsu”, “Sauvola” and “Wolf” algorithms for non-technical users.

This; what is Window size and how is it applied to output (or interface)? default value - window size: 200

I think the Window size is one of parameters of the filter (smoothing algorithm is an implementation of a digital image processing filter). Usually it is a size in pixels of a square which is used in process of filtering. With easy words, it's a size of a blur in pixels.
